contracts/EvmGolf.sol

```diff
@@ -0,0 +1,71 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+contract EvmGolf {
+    mapping(string => address) private _levels; // level name -> level address
+    mapping(address => uint) private _authors; // author -> level address
+    mapping(address => address) private _solutions; // level -> solution
+    mapping(address => address) private _records; // level -> author
+    mapping(address => uint) private _victories; // author -> number of victories
+
+    function registerLevel(address newLevel) external returns (bool) {
+        const levelABI = [ "function name() external returns (string memory)"];
+        const level = new ethers.Contract(newLevel, levelABI, ethers.getDefaultProvider());
```

Comment:
Ethers does not exist in Solidity. It's just a library that helps you talk to contracts from Javascript.

Reply:
I was using this to get the level contract. I should rename the variable, but do you mean I can't get the contract this way?

Reply:
The variable type for the level should be ILevel. And you alread have the address, so that should be enough to interact with the level from the main contract.